### DISCUSSION TITLE
Representing support of events

### DISCUSSION THREAD

**Discussion by foolip** [mdn/mdn-community#533] [Content]
This is a discussion thread to go with https://github.com/mdn/browser-compat-data/issues/7545, since sorting out event support in BCD will request changes to MDN as well.

Summary of what I think: group all `onfoo` event handler properties where they appear in documentation and in BCD tables, and treat them as a subfeature of the events they are for. Somehow.

To use a similar example to illustrate the problem, consider the "playing" event for media element, split like this in MDN:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/playing_event
https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onplaying

IMHO this is not a useful split, and including information about `onplaying` in the first page would be better.

Beyond that, showing the support for the event is tricky and is where BCD and MDN will interact on this. Showing support for the event and the `onfoo` attribute as two completely separate features is at best redundant, but any time the `onfoo` attribute was introduced too early or too late (very common) it becomes hard to interpret a split representation.

Many events also have their own event interfaces, such as [`AnimationEvent`])(https://developer.mozilla.org/en-US/docs/Web/API/AnimationEvent), but I believe those should remain separate, because often there are multiple event types for a single interfaces, in this case "animationstart", "animationiteration", "animationend",  and "animationcancel".

**Comment by pepelsbey** [comment]
Closing as answered/settled: [Yes, I'm very happy with the outcome in BCD, closing now](https://github.com/mdn/browser-compat-data/issues/7545#issuecomment-1721111675).
